### ISSUE TITLE
Remove data-activeid attributes and visible reactive variable placeholders

### ISSUE THREAD

**Issue by bob2517** [Active-CSS/active-css#37]
Investigating an potential option for removing the data-activeid attribute that is used a lot internally.

Gonna come off the tests to do this now, as this should improve performance as well as getting rid of something pretty annoying. Shouldn't take too long to implement if it works. Half way through the command tests is a good enough point to break off to do this.

**Comment by bob2517** [comment]
Looking good so far. Just need to sort out the new "<" which relies heavily on the data-active attribute. Need to rewrite that completely, but the idea was successful. No more data-activeid attribute! This is good...

**Comment by bob2517** [comment]
I'm going to try to get rid of the /* variable-placeholder */ comment that is currently used for reactive variables within text nodes as part of this. It would be really cool. I think I can just set up empty nodes and use a mapping technique similar to the data-activeid removal, although I won't know for sure if this will work until I have a go at it. This should be a bit easier to write. There were a couple of gotchas with removing the data-activeid, but this has less implications.

That would be really cool - that will be all the intrusive DOM additions removed if this works.

**Comment by bob2517** [comment]
That's reactive comment removal is all done and works nicely. The comment is there until the variable is populated, which I think is fine. If there's an error it's good to see what is going on in the place where it's supposed to go on.

Lastly is just the removal of the /* variable-placeholder */ comment wrappers around inline style reactive variables and that should be it. That'll require a slightly different tactic.

**Comment by bob2517** [comment]
Works like a charm!

Ok, this issue is complete and will go live with 2.4.0.

**Comment by bob2517** [comment]
Re-approaching this - I missed out removing variable type "active-var-i10HOSTlen" used for shadow DOM host referencing.

**Comment by bob2517** [comment]
The style tag is now working correctly offline. I've refactored the text node and style tag updates to make it more direct and a bit faster by using direct node reference rather than indirect activeid reference. But I've had to leave the attribute update as it was. It would have been nicer to use a treewalker with an attribute filter, but for some reason an attribute filter on a treewalker is deprecated.
This will get put up onto the 2.4.0 branch tomorrow. Trying to wrap up the clean-up issue to go up with this too so the core is bug-free as of current issues so I can get back onto the test suites and see what comes up next. Looking good though.
